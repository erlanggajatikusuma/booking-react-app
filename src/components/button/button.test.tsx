import React from 'react';
import renderer from 'react-test-renderer';
import {Button} from './Button';
import {Image} from 'react-native';

// Mock the Image component
jest.mock('react-native/Libraries/Image/resolveAssetSource', () => ({
  default: (source: any) => {
    return source;
  },
}));

// Mock the require statement for the image source
jest.mock('./image.png', () => 123);

describe('Button Component', () => {
  test('renders correctly with Text as children', () => {
    const tree = renderer
      .create(<Button label="Click me" onPress={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with link preset', () => {
    const tree = renderer
      .create(<Button label="Click Me" onPress={() => {}} preset="link" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with Image as children', () => {
    const tree = renderer
      .create(
        <Button onPress={() => {}}>
          <Image source={require('./image.png')} />
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('calls onPress when clicked', () => {
    const onPressMock = jest.fn();
    const tree = renderer.create(
      <Button label="Click Me" onPress={onPressMock} />,
    );

    // Access the root of the rendered component
    const rootInstance = tree.root;

    // Find the button element by its props
    const buttonElements = rootInstance.findAllByProps({label: 'Click Me'});

    // Simulate a button click on the first button found
    buttonElements[0].props.onPress();

    // Expect that the onPressMock function was called
    expect(onPressMock).toHaveBeenCalled();
  });

  it('renders correctly with disabled state', () => {
    const tree = renderer
      .create(<Button label="Click Me" disabled={true} onPress={() => {}} />)
      .toJSON();

    // Check if the tree is not null before making assertions
    expect(tree).not.toBeNull();

    // You can now safely make assertions on the tree
    expect(tree).toMatchSnapshot();
  });
});
