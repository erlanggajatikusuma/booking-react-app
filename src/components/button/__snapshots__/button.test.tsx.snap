// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly with Image as children 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6C5ECF",
      "borderRadius": 6,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 12,
    }
  }
>
  <Image
    source={123}
  />
</View>
`;

exports[`Button Component renders correctly with Text as children 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6C5ECF",
      "borderRadius": 6,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
          "paddingHorizontal": 12,
        },
        undefined,
      ]
    }
  >
    Click me
  </Text>
</View>
`;

exports[`Button Component renders correctly with disabled state 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "gray",
      "borderRadius": 6,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
          "paddingHorizontal": 12,
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`Button Component renders correctly with link preset 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "flex-start",
      "backgroundColor": "transparent",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 0,
      "paddingVertical": 0,
    }
  }
>
  <Text
    style={
      [
        {
          "borderBottomWidth": 1,
          "borderColor": "#6C5ECF",
          "color": "#6C5ECF",
          "fontSize": 16,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;
